docs/backend/api-definitions-guidelines.md

```diff
@@ -0,0 +1,58 @@
+# API Definitions Guidelines
+
+This document describes how service APIs will be managed for `InstructLab` and the sub-components of the `InstructLab` backend.
+
+## What parts of InstructLab need service APIs?
+
+There are two primary classes of service APIs needed to support InstructLab:
+
+* `Platform APIs`: These are APIs that are ignorant of `InstructLab` and provide generic AI platform capabilities (e.g., Fine Tuning, SDG, Eval)
+* `InstructLab APIs`: These are the APIs that reflect the user-facing functionality of `InstructLab` itself. They are aware of the end-to-end `InstructLab` workflow.
+
+The `InstructLab APIs` are essential for hosting `InstructLab` as a service in a repeatable way. The `Platform APIs` are critical for component reuse and extensibility (e.g., new SDG algorithms for new taxonomy data types), but they are not strictly required for hosting `InstructLab` as a service.
+
+## How will service APIs be defined?
+
+Service APIs will be defined using [OpenAPI](https://www.openapis.org/) format in [YAML](https://yaml.org/).
+
+## Where will service API definitions live?
+
+Service API definitions will live a new repository github.com/instructlab/service-api-definitions. This repo will have two primary responsibilities:
+
+1. House the static service API definitions
+2. Build and publish any language-specific generated packages for consumption by service implementation projects (see below)
+
+## How will service implementations reference shared APIs?
+
+When a project chooses to implement one or more service APIs, there are three acceptable methods for doing so, listed in order of preference:
+
+1. Consume a supported language-specific package. The `service-api-definitions` repo will build consumable packages with generated code for supported languages. This is the preferred method of consumption as it avoids repository references and code duplication.
```

Comment:
Good question. I'm coming from the `python` POV where `build` would be creating the `wheel` and pushing to PyPI. I haven't used `go` extensively since the repo-management was all centered around pulling the tip of `master`, so I'm not clear if there's any equivalent way to have statically prebuilt packages with `go.mod`. I can rephrase to be a little more generic.

Reply:
Go module URIs are git repos. I am not aware of package index for go modules. AFAIU some automation in the golang module repo will have to track changes to schema and generate commits to update bindings module. This should be doable. The comment of mine here is quite minor - the `build` term is not applicable to some platforms. But the essence - auto-generation of bindings - is the same.

Reply:
Makes sense (I'm _very_ outdated on `go`). I updated the language to try to fuzz the operation beyond strictly `build`.